src/MacVim/MMWindowController.h

```diff
@@ -19,30 +19,32 @@
 
 @interface MMWindowController : NSWindowController<NSWindowDelegate>
 {
-    MMVimController     *vimController;
-    MMVimView           *vimView;
-    BOOL                setupDone;
-    BOOL                windowPresented;
-    BOOL                shouldResizeVimView;
-    BOOL                shouldRestoreUserTopLeft;
-    BOOL                shouldMaximizeWindow;
-    int                 updateToolbarFlag;
-    BOOL                keepOnScreen;
-    NSString            *windowAutosaveKey;
-    BOOL                fullScreenEnabled;
-    MMFullScreenWindow  *fullScreenWindow;
-    int                 fullScreenOptions;
-    BOOL                delayEnterFullScreen;
-    NSRect              preFullScreenFrame;
-    MMWindow            *decoratedWindow;
-    NSString            *lastSetTitle;
-    int                 userRows;
-    int                 userCols;
-    NSPoint             userTopLeft;
-    NSPoint             defaultTopLeft;
-    NSToolbar           *toolbar;
-    BOOL                resizingDueToMove;
-    int                 blurRadius;
+    MMVimController               *vimController;
+    MMVimView                     *vimView;
+    BOOL                          setupDone;
+    BOOL                          windowPresented;
+    BOOL                          shouldResizeVimView;
+    BOOL                          shouldRestoreUserTopLeft;
+    BOOL                          shouldMaximizeWindow;
+    int                           updateToolbarFlag;
+    BOOL                          keepOnScreen;
+    NSString                      *windowAutosaveKey;
+    BOOL                          fullScreenEnabled;
+    MMFullScreenWindow            *fullScreenWindow;
+    int                           fullScreenOptions;
+    BOOL                          delayEnterFullScreen;
+    NSRect                        preFullScreenFrame;
+    MMWindow                      *decoratedWindow;
+    NSString                      *lastSetTitle;
+    int                           userRows;
+    int                           userCols;
+    NSPoint                       userTopLeft;
+    NSPoint                       defaultTopLeft;
+    NSToolbar                     *toolbar;
+    BOOL                          resizingDueToMove;
+    int                           blurRadius;
+    NSMutableArray<NSOperation *> *afterWindowPresentedQueue;
+    NSLock                        *afterWindowPresentedQueueLock;
```

Comment:
If we really need lock here, it means that some code path calls `handleShowDialog:` on the other thread. It is supposed to be called on the main thread only. We should fix it.